config/m4/module.m4

```diff
@@ -36,25 +37,32 @@ m4_define([_CHECK_MODULE_UNDEFVAR],
           ])
 
 #
-# Init progress
+# Init progress: set 'progress' variable to one of values yes/no/guess/skip/dir
+#                based on 'mod-dir' value. In case of 'dir' value is set then
+#                also set dir-int, inc-flag, dir-lib and dir-flag variables or call
+#                'if-failed' macro if some of directories are missing
 #
 # Usage: _CHECK_MODULE_INIT_PROGRESS([mod-dir], [progress],
 #                                    [dir-int], [inc-flag], [dir-lib], [dir-flag], [if-failed])
 #
 m4_define([_CHECK_MODULE_INIT_PROGRESS],
-          [
-            _CHECK_MODULE_IS_YNG([$1],
-                                 [$2=yes],   # yes
-                                 [$2=skip],  # no
-                                 [$2=guess], # guess
-                                 [$2=skip],  # empty
-                                 [$2=dir     # DIR
-                                  _CHECK_MODULE_SET_DIR([$1], [$3], [$4], [$5],
-                                                        [$6], [$7])])
-          ])
+         [AS_CASE(["x$1"],
+                  ["xyes"],   [$2=yes],
+                  ["xno"],    [$2=skip],
+                  ["xguess"], [$2=guess],
+                  ["x"],      [$2=skip],
+                              [$2=dir
+                               _CHECK_MODULE_SET_DIR([$1], [$3], [$4], [$5], 
+                                                     [$6], [$7])])
+         ])
 
 #
-# Configure module directories
+# Configure module directories:
+#     inc=$mod-dir/include
+#     incflag=-I$mod-dir/include
+#     lib=$mod-dir/lib64 or $mod-dir/lib
+#     libflag=-L$lib (from previous step)
+#     if any of directories is missing - call if-failed macro
```

Comment:
1. Do we really need to set "inc" and "lib"?
I would expect only  "incflag" and "libflag" would be needed

2. I think the naming of variables should follow existing conventions to make it easier to maintain.
For example: set MOD_LDFLAGS or ${MODULE}_LDFLAGS instead of "libflag"